.github/workflows/build.yml

```diff
@@ -28,33 +22,35 @@ jobs:
       - name: Update Fastlane
         run: bundle update fastlane
 
+        # Required for KotlinMultiplatform
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '11'
+
       - name: Pod Install
         run: bundle exec fastlane cru_shared_lane_pod_install
 
       - name: Run Tests
         run: bundle exec fastlane cru_shared_lane_run_tests
 
       - name: Download And Commit Latest OneSky Localizations
-        if: github.event_name == 'workflow_dispatch'
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
         run: bundle exec fastlane cru_shared_lane_download_and_commit_latest_one_sky_localizations
 
-      - name: Decode AppleAppStoreApi.json.enc For Fastlane App Store Connect Access 
-        if: github.event_name == 'workflow_dispatch'
+      - name: Decrypt AppleAppStoreApi.json.enc For Fastlane App Store Connect Access 
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
         env: 
           KEY: ${{ secrets.ENCRYPTED_CF3A694B4ADE_KEY }}
           IV: ${{ secrets.ENCRYPTED_CF3A694B4ADE_IV }}
         run: openssl aes-256-cbc -K ${KEY} -iv ${IV} -in fastlane/AppleAppStoreApi.json.enc -out fastlane/AppleAppStoreApi.json -d
 
       - name: Increment Xcode Project Build Number
-        if: github.event_name == 'workflow_dispatch'
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
         run: bundle exec fastlane cru_shared_lane_increment_xcode_project_build_number
```

Comment:
why did you create new fastlane lanes instead of using the ones we had?

Reply:
Hey @frett there are a number of reasons.

Reply:
@frett  I'll gather some notes and include them here.

Reply:
Hey @frett the biggest reason was how code signing and match was done, the cru fastfile is including MPDX's Cal Directory Extension in code signing and match which isn't necessary and would rather opt for a more dynamic approach to this.  That way if we have iOS apps that include other app extensions we don't bloat the cru fastfile.  Also with match you don't need to create a new match for every app extension.  Match supports multiple bundle ids.

Reply:
@frett also some of the naming, for example, in localizations download they're using ```CRU_SCHEME``` which actually has nothing to do with the localizations download, that is actually a root directory name.

Reply:
@frett I do want to look into including a firebase distribution in the cru fastfile as well which would require different code signing than what is used for TestFlight, which is another reason I wanted to break up some of the lanes.

Reply:
I definitely agree about not including the MPDX call directory stuff in the shared fastlane file, I hadn't realized a contractor had done that several years ago. I would have definitely pushed more a more configurable option, or at least putting some call directory specific lanes in the MPDX Fastfile.

Reply:
ok, I'm going to process the new lanes a bit, it would be nice if there was a PR with the new lanes so I can leave some comments there

Reply:
@frett just created a PR for those new lanes